content/docs/rfcs/64/README.md

```diff
@@ -0,0 +1,258 @@
+to allow the underlying node to automatically select the target shard on the Waku Network. See the section on [autosharding](#autosharding) for more.
+
+## RLN rate-limiting
+
+The [17/WAKU2-RLN-RELAY](https://rfc.vac.dev/spec/17/) network uses [32/RLN-V1](https://rfc.vac.dev/spec/32/) proofs
+to ensure that a pre-agreed rate limit is not exceeded by any publisher.
+While the network is under capacity,
+individual relayers MAY choose to freely route messages without RLN proofs
+up to a discretionary bandwidth limit after which messages without proofs MUST be discarded.
+This implies that quality of service and reliability is significantly lower for messages without proofs
+and at times of high network utilization these messages may not be relayed at all.
+
+### RLN Parameters
+
+For the Waku Network,
+the `epoch` is set to `1` second
+and the maximum number of messages published per `epoch` is limited to `1` per publisher.
+The `max_epoch_gap` is set to `20` seconds,
+meaning that validators MUST _ignore_ messages with an `epoch` more than 20 seconds into the past or future compared to the validator's own clock.
+
+### Memberships
+
+Each publisher to the Waku Network SHOULD register an RLN membership
+with contract (TBD)[contract address]
+using the method described in (TBD)[how to register membership....]
+
+### RLN Proofs
+
+(TBD)
+
+## Network traffic
+
+All payload on the Waku Network MUST be encapsulated in a [14/WAKU2-MESSAGE](https://rfc.vac.dev/spec/14/)
+with rate limit proof extensions defined for [17/WAKU2-RLN-RELAY](https://rfc.vac.dev/spec/17/#payloads).
+Each message on the Waku Network SHOULD be validated by each relayer,
+according to the rules discussed under [message validation](#message-validation).
+
+### Message Attributes
+
+- The mandatory `payload` attribute MUST contain the message data payload as crafted by the application.
+- The mandatory `content_topic` attribute MUST specify a string identifier that can be used for content-based filtering. This is also crafted by the application. See dhuhfh for more on the content topic format.    
+- The optional `meta` attribute MAY be omitted. If present this will form part of the message uniqueness vector described in [14/WAKU2-MESSAGE](https://rfc.vac.dev/spec/14/).
+- The optional `version` attribute SHOULD be set to `0`. It MUST be interpreted as `0` if not present.
+- The mandatory `timestamp` attribute MUST contain the Unix epoch time at which the message was generated by the application. The value MUST be in nanoseconds. It MAY contain a fudge factor of up to 1 seconds in either direction to improve resistance to timing attacks.
+- The optional `ephemeral` attribute MUST be set to `true` if the message should not be persisted by the Waku Network.
+- The optional `rate_limit_proof` attribute SHOULD be populated with the RLN proof as set out in dhjfhjhd. Messages with this field unpopulated MAY be discarded from the network by relayers.
```

Comment:
dhjfhjhd. --> probably TBD or link to a section?

Reply:
Hahaha. Bad proofreading. Fixed in [74ba3ef](https://github.com/vacp2p/rfc/pull/616/commits/74ba3efb1dbc338ecccbc40276db2ba561dff68f). Thanks!